src/transforms/at-current-range.js

```diff
@@ -95,70 +95,11 @@ export function _delete(transform) {
 
 export function deleteBackward(transform, n = 1) {
   const { state } = transform
-  const { document, selection } = state
-  let after
-
-  const { startKey } = selection
-  const startNode = document.getDescendant(startKey)
-
-  if (selection.isExpanded) {
-    after = selection.collapseToStart()
-  }
-
-  else if (selection.isAtStartOf(document)) {
-    after = selection
-  }
-
-  else if (selection.isAtStartOf(startNode)) {
-    const previous = document.getPreviousText(startNode)
-    const prevBlock = document.getClosestBlock(previous)
-    const prevInline = document.getClosestInline(previous)
-
-    if (prevBlock && prevBlock.isVoid) {
-      after = selection
-    } else if (prevInline && prevInline.isVoid) {
-      const prevPrev = document.getPreviousText(previous)
-      after = selection.collapseToEndOf(prevPrev)
-    } else {
-      after = selection.collapseToEndOf(previous)
-    }
-  }
-
-  else if (selection.isAtEndOf(startNode) && startNode.length == 1) {
-    const block = document.getClosestBlock(startKey)
-    const highest = block.getHighestChild(startKey)
-    const previous = block.getPreviousSibling(highest)
-    const next = block.getNextSibling(highest)
-
-    if (previous) {
-      if (previous.kind == 'text') {
-        if (next && next.kind == 'text') {
-          after = selection.merge({
-            anchorKey: previous.key,
-            anchorOffset: previous.length,
-            focusKey: previous.key,
-            focusOffset: previous.length
-          })
-        } else {
-          after = selection.collapseToEndOf(previous)
-        }
-      } else {
-        const last = previous.getLastText()
-        after = selection.collapseToEndOf(last)
-      }
-    } else {
-      after = selection.moveBackward(n)
-    }
-  }
-
-  else {
-    after = selection.moveBackward(n)
-  }
```

Comment:
Yes, it is because most of the work is already done in the operations themselves. It may be a bit slower in the end, but it's much more convenient to write transforms this way. One of the great idea from @SamyPesse